server/web/services/submission_service.py

```diff
@@ -4,11 +4,14 @@
 
 from .. import database
 from ..models.submission_model import Submission
+from ..models.environment_model import Environment
+from ..models.game_model import Game
 
-def evaluate_agent(agent):
-    env = gym.make("FrozenLake-v0")
+def evaluate_agent(agent, game_id):
+    env_name = Environment.from_dict(list(database.find_env_name_by_game_id(game_id, show_fields=["gym_env"]))[0]).gym_env
+    env = gym.make(env_name)
 
-    episodes = 10000
+    episodes = int(Game.from_dict(list(database.find_single_game(game_id))[0]).num_of_episodes)
```

Comment:
- Use the method find_game in game_service.py.
- Change the game model such that the num_of_episodes will be an integer and remove int() casting from this line.